src/cmclient/cmclient.hpp

```diff
@@ -107,12 +111,16 @@ class CMClient : public aos::sm::launcher::InstanceStatusReceiverItf,
     aos::sm::launcher::LauncherItf*                              mLauncher = {};
     ResourceManager*                                             mResourceManager = {};
     DownloadReceiverItf*                                         mDownloader = {};
-    aos::Thread<>                                                mThread = {};
+    aos::Thread<>                                                mThreadWriter = {};
+    aos::Thread<>                                                mThreadReader = {};
     atomic_t                                                     mState = {};
-    vch_handle                                                   mSMVChanHandler;
+    bool                                                         mWriteFail = {};
```

Comment:
It is not good the some internal states are handled in different places. We have `mState` add additional field there such a `eWriteFail`. Also `eConnected` is not used anymore - remove it.

Reply:
Done